src/compiler/checker.ts

```diff
@@ -16052,9 +16176,13 @@ namespace ts {
                     if (produceDiagnostics) {
                         const symbol = getNodeLinks(node).resolvedSymbol;
                         const typeParameters = symbol.flags & SymbolFlags.TypeAlias ? getSymbolLinks(symbol).typeParameters : (<TypeReference>type).target.localTypeParameters;
-                        checkTypeArgumentConstraints(typeParameters, node.typeArguments);
+                        const minTypeArgumentCount = symbol.flags & SymbolFlags.TypeAlias ? getSymbolLinks(symbol).minTypeArgumentCount : (<TypeReference>type).target.minTypeArgumentCount;

```

Comment:
This is getting the pre-computed count of non-optional type arguments. As with `typeParameters` above, this is stored on `SymbolLinks` for a type alias, or the generic target for a `TypeReference`.